app/presenters/speech_presenter.rb

```diff
@@ -0,0 +1,39 @@
+class SpeechPresenter < ContentItemPresenter
+  include DateHelper
+
+  def delivery_type
+    if content_item.document_type == "authored_article"
+      I18n.t("formats.speech.written_on")
+    else
+      I18n.t("formats.speech.delivered_on")
+    end
+  end
+
+  def delivered_on_metadata
+    "#{delivered_on}#{speech_type_explanation}"
+  end
+
+  def important_metadata
+    { I18n.t("formats.speech.location") => location, delivery_type => delivered_on_metadata }
+  end
```

Comment:
You could probably accept view_context as a parameter to this method (and pass it on to delivered_on_metadata) which would allow you to not have the view_context method. (What do you think, @leenagupte ?)

Reply:
Is `view_context` really needed? Can `tag.time` not be called in the view?

I think perhaps that the `delivered_on` method needs to be looked at again.

It's modelling `delivered_on_date` which could be moved to the model, and then calling a helper method on it, where as I the functionality in the `important_metadata` and `delivered_on_metadata` could just be moved to the view.

Reply:
important_metadata is only called in one place (a view) - it could potentially be inlined there?